.github/workflows/image.yml

```diff
@@ -11,20 +11,51 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: write
+      packages: write
     steps:
       - name: Checkout code
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       - name: Set environment variables
         run: echo "GIT_BRANCH=${GITHUB_REF##*/}" >> $GITHUB_ENV
-      - name: Install buildx
-        id: buildx
-        uses: crazy-max/ghaction-docker-buildx@v1
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
         with:
-          version: latest
-      - name: Login to docker hub
-        run: echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u "${{ secrets.DOCKER_USERNAME }}" --password-stdin
-      - name: Build the image
-        shell: bash
-        run: ./.github/actions/build.sh
-        env:
-          VERSION: ${{ inputs.version }}
\ No newline at end of file
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Generate Docker metadata
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            name=zurdi15/romm
+            name=ghcr.io/zurdi15/romm
+          flavor: |
+            # latest on master branch, prefix dev on other branches
+            latest=true
+            prefix=${{ github.ref != format('refs/heads/{0}', 'master') && 'dev-' || '' }},onlatest=true
+          tags: |
+            type=raw,value=${{ inputs.version }}
+          labels: |
+            org.opencontainers.image.version=${{ inputs.version }}
+      - name: Build image
```

Comment:
Is there any possibility here to just build for arm64 in every branch except master as I am developing in a RPi and at least for me doesn't make sense to build for amd64 in developing branches?

Reply:
Yes, you can copy the same code for github.ref and apply it in here.
https://github.com/docker/build-push-action#inputs

```yaml
platforms: |
  linux/arm64
  ${{ github.ref == format('refs/heads/{0}', 'master') && 'linux/amd64' || '' }}
```

However, the build time is going to be slow either way because of `buildx` for `arm64` which you are interested in.
There is no problem in building both, then any machine (RPi or computer) will only download its platform layers only. So I would keep the current build.

Reply:
Ohh okay, I see. I keep then the current code. Thank you for the explanation